include/swift/AST/DeclContext.h

```diff
@@ -802,6 +802,12 @@ class IterableDeclContext {
   // Some Decls are IterableDeclContexts, but not all.
   static bool classof(const Decl *D);
 
+  /// Return a hash of all tokens in the body for dependency analysis.
+  /// Empty string means no hash available.
+  StringRef getBodyFingerprint() const;
```

Comment:
I'm not a fan of overloading the meaning of the empty string in APIs like this.  I think this should return an `Optional<StringRef>` and clients should assert it's not ever the empty string.

Reply:
Nice catch! I've changed it to an optional, and also changed the \c StringRef to a \c std::string in order to make the memory safety less brittle in the face of request evaluator evolution.